spec/acceptance/init_spec.rb

```diff
@@ -20,8 +20,35 @@ class { 'bolt': use_release_package => false }
     end
   end
 
-  describe command('yum clean all --verbose; rm -rf /etc/yum.repos.d/puppet-tools-release.repo; yum erase --assumeyes puppet-bolt puppet-tools-release') do
-    its(:exit_status) { is_expected.to eq 0 }
+  describe 'with use_release_package=false & ensure=absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'bolt':
+          use_release_package => false,
+          version => 'absent',
+        }
+        PUPPET
+      end
+    end
+    # rubocop:disable RSpec/RepeatedExampleGroupBody
+    describe package('puppet-bolt') do
+      it { is_expected.not_to be_installed }
+    end
+
+    describe package('puppet-tools-release') do
+      it { is_expected.not_to be_installed }
+    end
+    # rubocop:enable RSpec/RepeatedExampleGroupBody
+
+    # puppet 7 and below remove repos from the file, puppet 8 deletes the file
+    describe file('/etc/yum.repos.d/puppet-tools.repo') do
+      if Gem::Version.new(fact('puppetversion')) < Gem::Version.new('8.0.0')
+        it { expect(file('/etc/yum.repos.d/puppet-tools.repo')).to be_file.or(have_attributes(size: 0)) }
```

Comment:
The size 0 check is skipped. Did you mean
```suggestion
        it { expect(file('/etc/yum.repos.d/puppet-tools.repo')).to be_file.and(have_attributes(size: 0)) }
```